csharp-basics/exercises/Arrays/Arrays/TicTacToe/Program.cs

```diff
@@ -5,32 +5,123 @@ namespace TicTacToe
+
+                    DisplayBoard();
+
+                    if (CheckWin(currentPlayer))
+                    {
+                        Console.WriteLine($"'{currentPlayer}' wins!");
+                        gameOver = true;
+                    }
+                    else if (CheckTie())
+                    {
+                        Console.WriteLine("The game is a tie.");
+                        gameOver = true;
+                    }
+                    else
+                    {
+                        currentPlayer = currentPlayer == 'X' ? 'O' : 'X';
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("That spot is already taken. Try again.");
+                }
+            }
+
+            Console.ReadLine();
         }
 
         private static void InitBoard()
         {
-            // fills up the board with blanks
-            for (var r = 0; r < 3; r++)
+            for (int r = 0; r < 3; r++)
             {
-                for (var c = 0; c < 3; c++)
+                for (int c = 0; c < 3; c++)
+                {
                     board[r, c] = ' ';
+                }
             }
-            
         }
 
         private static void DisplayBoard()
         {
-            Console.WriteLine("  0  " + board[0, 0] + "|" + board[0, 1] + "|" + board[0, 2]);
-            Console.WriteLine("    --+-+--");
-            Console.WriteLine("  1  " + board[1, 0] + "|" + board[1, 1] + "|" + board[1, 2]);
-            Console.WriteLine("    --+-+--");
-            Console.WriteLine("  2  " + board[2, 0] + "|" + board[2, 1] + "|" + board[2, 2]);
-            Console.WriteLine("    --+-+--");
+            Console.WriteLine("   0   1   2");
+            Console.WriteLine("  +---+---+---+");
+            for (int r = 0; r < 3; r++)
+            {
+                Console.Write($"{r} ");
+                for (int c = 0; c < 3; c++)
+                {
+                    Console.Write($"| {board[r, c]} ");
+                }
+                Console.Write("|");
+                Console.WriteLine();
+                Console.WriteLine("  +---+---+---+");
+            }
+        }
+
+        private static bool CheckWin(char player)
+        {
+            for (int r = 0; r < 3; r++)
+            {
+                if (board[r, 0] == player && board[r, 1] == player && board[r, 2] == player)
+                {
+                    return true;
+                }
+            }
+
+            for (int c = 0; c < 3; c++)
+            {
+                if (board[0, c] == player && board[1, c] == player && board[2, c] == player)
+                {
+                    return true;
+                }
+            }
+
+            if (board[0, 0] == player && board[1, 1] == player && board[2, 2] == player)
+            {
+                return true;
+            }
+
+            if (board[2, 0] == player && board[1, 1] == player && board[0, 2] == player)
```

Comment:
Tev atgriež vienu un to pašu iekš if, tad var likt kopā caur ||.

Un tieši šādu kodu es jau redzēju..

Reply:
Pārtaisīju uzdevumu.